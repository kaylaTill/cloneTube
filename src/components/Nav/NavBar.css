.main-navbar {
  display: flex;
  align-items: center;
  height: 45px;
  background-color: #ffffff;
}

.youtube-logo {
  margin-left: 2.5em;
  width: 10%;
  height: 10%;
}

.menu-icon {
  position: relative;
  right: 5rem;
  height: 18px;
}


.bell-icon {
  position: relative;
  height: 18px;
}


@media screen and (max-width: 500px) {
  .youtube-logo {
    margin-left: 5%;
  }

  .search-wrapper {
    width: 1000%;
  }
}
