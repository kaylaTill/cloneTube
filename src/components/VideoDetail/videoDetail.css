div {
  font-family: Arial, sans-serif;
}


.video-detail {
  display: flex;
  width: 85%;
  margin-left: calc(94px + 4.5em + 80px);
}

.thumbnail {
  border: none;
  background-size: cover;
}

.video-player {
  width: 115%;
  height: 430px;
  max-width: 854px;
  max-height: 490px;
  display: block;
  margin: 2em auto;
  border: none;
}

.video-list-title {
  color: #000000;
  display: block;
  margin: 0 0 4px 0;
  max-height: 3.2rem;
  overflow: hidden;
  font-size: 0.9rem;
  font-weight: 500;
  line-height: 1.6rem;
  font-weight: bold;
}

.channelName {
  color: #000000;
  display: inline-block;
}

.main-video-title {
  color: #000000;
  font-weight: 400;
  margin-bottom: 0;
  font-weight: bold;
}

.main-video-description {
  color: #000000;
}

.text-container {
  display: flex;
  flex-direction: column;
  padding-left: 10px;
}

.video-detail-wrapper {
  max-width: 854px;
  max-height: 490px;
}

.main-video-channelTitle {
  color: #000000;
}


@media screen and (max-width: 1580px) {
  .video-detail {
    width: 95%;
    margin-left: 0;
    margin: auto;
  }
  .video-detail-wrapper {
    min-width: 640.5px;
    min-height: 362.25px;
  }
  .video-player {
    max-width: 115%;
    max-height: 500px;
  }

  .video-list-wrapper {
    width: 70%;
    padding-left: 0;
    max-height: 400%;
  }

  .list-item {
    margin: 20px 0 20px 20px;
    height: 10%;
    color: #747474;
  }
}

@media (max-width: 957px) {
  .video-detail {
    width: 90%;
    flex-direction: column;
    margin-left: 0;
    margin: auto;
  }
  .video-detail-wrapper {
    width: 100%;
    max-height: none;
    min-width: 0;
    color: #747474;
  }
  .video-player {
    max-width: none;
    min-width: 0;
    width: 100%;
  }

  .video-list-wrapper {
    width: 70%;
    padding-left: 0;
  }

  .search-wrapper {
    width: 100%;
    margin-left: 2em;
  }

  .search-bar {
    width: 94%;
  }
}
