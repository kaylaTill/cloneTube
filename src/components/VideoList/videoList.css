
.next {
  position: relative;
  color: #000000;
  float: left;
  font-size: 14px;
  margin-top: 25px;
  left: 7.5rem;
}

.autoplay {
  position: relative;
  color: #787878;
  font-size: 11px;
  float: left;
  margin-top: 14px;
  left: 9rem;
  font-weight: bold;
}

.switch {
  float: left;
  position: relative;
  left: 10rem;
}


.react-switch-checkbox {
  height: 0;
  width: 0;
  float: left;
  visibility: hidden;
}

.react-switch-label {
  display: flex;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  width: 27px;
  height: 10px;
  background-color: #cacaca;
  border-radius: 50px;
  position: relative;
  margin-top: 15px;
}

.react-switch-label .react-switch-button {
  content: '';
  position: absolute;
  left: 2px;
  width: 15px;
  height: 15px;
  border-radius: 20px;
  transition: 0.2s;
  background: #787878;
}

.react-switch-checkbox:checked + .react-switch-label .react-switch-button {
  left: calc(100% - 2px);
  transform: translateX(-100%);
}
.react-switch-label:active .react-switch-button {
  width: 60px;
}



.list-item {
  cursor: pointer;
  margin: 2em 0 0;
  display: flex;
  align-items: flex-start;
  color: rgba(0, 0, 0, 0.5);
}

.video-list-wrapper {
  position: relative;
  left: 15%;
  margin-top: 35px;
  background-color: #f8f8f8;
  max-height: 400%;
  overflow: scroll;
  overflow-x: hidden; 
  padding-right: 10px;
}
