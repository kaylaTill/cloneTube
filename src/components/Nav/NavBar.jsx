import React from "react";
import YoutubeLogo from "./icons/youtube-logo.png"
import MenuIcon from "./icons/icons8-squared-menu-24.png"
import NavIcon from "./NavIcon"
import NavBell from "./icons/icons8-bell-24.png"
import NavVideo from "./icons/icons8-video-call-24.png"
import UserPhoto from "./icons/userphoto.png"
import "./NavBar.css";



const NavBar = props => {
  return (
    <div className="menu">
      <nav className="main-navbar">
        <NavIcon/>
        <a href="/">
          <img src={YoutubeLogo} alt="Youtube Logo" className="youtube-logo" />
        </a>
        {props.children}
        <img src={NavVideo} alt="Video-icon" className="video-icon" ></img>
        <img src={MenuIcon} alt="Menu-icon" className="menu-icon" ></img>
        <img src={NavBell} alt="Bell-icon" className="bell-icon" ></img>
        <img src={UserPhoto} alt="User-icon" className="user-icon" ></img>
      </nav>
    </div>
  );
};

export default NavBar;
