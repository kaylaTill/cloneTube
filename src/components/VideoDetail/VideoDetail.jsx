import React from "react";
import "./videoDetail.css";

const VideoDetail = props => {
  if (!props.video) return null;

  let fulldate = props.video.publishedAt.split('-')
  let year = fulldate[0];
  let month = fulldate[1];
  let day = fulldate[2].slice(0, 2);
  const months = {
     "01" :'Jan',
      "02" :'Feb',
      "03" :'Mar',
      "04" :'Apr',
      "05" :'May',
      "06" :'Jun',
      "07" :'Jul',
      "08" :'Aug',
      "09" :'Sep',
      "01" :'Oct',
      "01" :'Nov',
     "01" :'Dec',
  };

  let videoId = props.video.id.videoId,
  url = `https://youtube.com/embed/${videoId}`;

  return (
    <div className="video-detail">
      <div className="video-detail-wrapper">
        <iframe src={url} title={`${props.video.title}`} className="video-player" />
        <h3 className="main-video-title">{`${props.video.title}`}</h3>
        <small className="main-video-date">
          {`${months[month]} ${day}, ${year}`}
        </small>
        <hr className="divider"></hr>
        <small className="main-video-channelTitle">
          {`${props.video.channelTitle}`}
        </small>
        <br />
        <small className="main-video-description">
          {`${props.video.description}`}
        </small>
        

      </div>
      {props.children}
    </div>
  );
};

export default VideoDetail;
