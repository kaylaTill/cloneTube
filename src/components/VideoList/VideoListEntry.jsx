import React from "react";
import "./videoList.css";

const VideoListItem = props => {
  const videoData = {
    title: props.snippet.title,
    thumbnail: props.snippet.thumbnails.default.url,
    channel: props.snippet.channelTitle
  };
  return (
    <div className="list-group-item">
      <li className="list-item" onClick={() => props.onVideoSelect(props)}>
        <img src={videoData.thumbnail} alt={videoData.title} className="thumbnail"/>
        <div className="text-container">
          <span className="video-list-title">{videoData.title}</span>
          <small className="channelName">{videoData.channel}</small>
        </div>
      </li>
    </div>
  );
};

export default VideoListItem;
