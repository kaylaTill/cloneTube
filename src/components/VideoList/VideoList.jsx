import React from 'react';
import './videoList.css';
import VideoListEntry from './VideoListEntry';
import Switch from "react-switch";


class VideoList extends React.Component {
  constructor(props) {
    super(props)
    this.state = { checked: false };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(checked) {
    this.setState({ checked });
  }
  
  render() {
    let videos = props.videos;
    return (
      <div>
        <div className="next">Up next</div>
        <div className="autoplay">
          <label>
            <span className="autoplay">Autoplay</span>
            <Switch onChange={this.handleChange} checked={this.state.checked} />
          </label>
        </div>
  
        <ul className="video-list-wrapper">
          {videos.map(video => <VideoListEntry {...video} key={video.etag} onVideoSelect={props.onVideoSelect} />)}
        </ul>
      </div>
    );
  }
};

export default VideoList;