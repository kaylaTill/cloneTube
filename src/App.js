import React, { Component } from 'react';
import _ from 'lodash';
import YouTubeSearch from 'youtube-api-search';
import Search from './components/SearchBar/SearchBar';
import VideoList from './components/VideoList/VideoList';
import VideoDetail from './components/VideoDetail/VideoDetail';
import NavBar from './components/Nav/NavBar';
import './App.css';
import API_KEY from './key';

class App extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      videos: [],
      selectedVideo: null,
      term: ''
    };
    this.getTerm = this.getTerm.bind(this);
  }

  componentDidMount() {
    this.searchVideoHandler(this.state.term); 
  }


  async getTerm(term) {
    await this.setState({term: term})
    this.searchVideoHandler(this.state.term)
  }

  searchVideoHandler = term => YouTubeSearch({ key: API_KEY, term }, videos => this.setState({ videos, selectedVideo: videos[0] }));

  render() {
    return (
      <div className="App">
        <NavBar> 
          <Search sendTerm={this.getTerm} />
        </NavBar>
        
        <VideoDetail videos={this.state.selectedVideo} > 
          <VideoList 
            videos={this.state.videos} 
            onVideoSelect={selectedVideo => this.setState({selectedVideo})} 
          /> 
        </VideoDetail>
      </div>
    );
  }
}

export default App;
