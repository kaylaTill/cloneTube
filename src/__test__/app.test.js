import React from 'react';
import App from '../App.js';
import NavBar from '../components/Nav/NavBar';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount } from 'enzyme';
import VideoDetail from '../components/VideoDetail/VideoDetail';
import VideoList from '../components/VideoList/VideoList';
Enzyme.configure({ adapter: new Adapter() });



describe('App', function () {

    
    it('should be a stateful class component', function () {
        expect(React.Component.isPrototypeOf(App)).toBe.true;
    });
    
    
    it('should render a single NavBar component', function () {
        const wrapper = mount(<App />);
        var nav = NavBar;
        // expect(wrapper.find(nav)).toExist();
        expect(wrapper.find(nav)).toMatchSnapshot()
    });
    
    it('should render a single VideoDetail component', function () {
        const wrapper = mount(<App />);
        var videoDetail = VideoDetail;
        expect(wrapper.find(videoDetail)).toMatchSnapshot()
    });
    
    
    it('should render a single VideoList component', function () {
        const wrapper = mount(<App />);
        var list = VideoList;
        expect(wrapper.find(list)).toMatchSnapshot()
    });
});