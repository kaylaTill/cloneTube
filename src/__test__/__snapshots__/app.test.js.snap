// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a single NavBar component 1`] = `ReactWrapper {}`;

exports[`App should render a single VideoDetail component 1`] = `ReactWrapper {}`;

exports[`App should render a single VideoList component 1`] = `ReactWrapper {}`;
